cpp/include/kvikio/shim/cufile.hpp

```diff
@@ -50,7 +50,7 @@ class cuFileAPI {
  private:
   cuFileAPI()
   {
-    void* lib = load_library({"libcufile.so.1", "libcufile.so.0", "libcufile.so"});
+    void* lib = load_library("libcufile.so.0");
```

Comment:
Suggestion from offline discussion is to do something like this, but ideally only under CUDA 11. Seems this shouldn't be an issue any more in CUDA 12

```suggestion
    void* lib = load_library({"libcufile.so.0", "libcufile.so"});
```

Reply:
Hard no.

Reply:
Discussed via Slack. This is an alternative that works for CUDA versions before 11.7.1 when this was fixed. (Also this passes clang-format.)

```suggestion
    // CUDA versions before CUDA 11.7.1 did not ship libcufile.so.0, so this is
    // a workaround that adds support for all prior versions of libcufile.
    void* lib = load_library({"libcufile.so.0",
                              "libcufile.so.1.3.0" /* 11.7.0 */,
                              "libcufile.so.1.2.1" /* 11.6.2, 11.6.1 */,
                              "libcufile.so.1.2.0" /* 11.6.0 */,
                              "libcufile.so.1.1.1" /* 11.5.1 */,
                              "libcufile.so.1.1.0" /* 11.5.0 */,
                              "libcufile.so.1.0.2" /* 11.4.4, 11.4.3, 11.4.2 */,
                              "libcufile.so.1.0.1" /* 11.4.1 */,
                              "libcufile.so.1.0.0" /* 11.4.0 */});
```

Reply:
Thanks @bdice!